File: android/src/main/java/com/emarsys/plugin/emarsys/EmarsysGeofence.java
package com.emarsys.plugin.emarsys;

import com.emarsys.Emarsys;
import com.emarsys.plugin.emarsys.utils.mappers.GeofenceMapper;
import org.json.JSONArray;

public class EmarsysGeofence {

    public void enable(EmarsysCompletionListener listener) {
        Emarsys.getGeofence().enable((error) -> listener.onComplete(error));
    }

    public void disable() {
        Emarsys.getGeofence().disable();
    }

    public boolean isEnabled() {
        return Emarsys.getGeofence().isEnabled();
    }

    public JSONArray getRegisteredGeofences() {
        return GeofenceMapper.mapGeofences(Emarsys.getGeofence().getRegisteredGeofences());
    }
}

File: ios/Sources/EmarsysPlugin/EmarsysGeofence.swift
import Foundation
import EmarsysSDK

@objc public class EmarsysGeofence: NSObject {
    @objc public func enable(completion: @escaping (Error?) -> Void) {
        Emarsys.geofence.enable { error in completion(error) }
    }

    @objc public func disable() {
        Emarsys.geofence.disable()
    }

    @objc public func isEnabled() -> Bool {
        return Emarsys.geofence.isEnabled()
    }

    @objc public func getRegisteredGeofences() -> [[String: Any]] {
        return Emarsys.geofence.registeredGeofences().map { GeofenceMapper.map($0) }
    }
}
